std/experimental/allocator/common.d

```diff
@@ -743,3 +743,147 @@ version(unittest)
         }
     }
 }
+
+/+
+Can the representation be determined at compile time to consist of nothing but
+zero bits? Padding between a struct's fields is not considered.
++/
+private template isAllZeroBits(T, T value)
+{
+    import std.traits : isDynamicArray, isStaticArray;
+    static if (isDynamicArray!(typeof(value)))
+        enum isAllZeroBits = value is null && value.length == 0;
+    else static if (is(typeof(value is null)))
+        enum isAllZeroBits = value is null;
+    else static if (is(typeof(value is 0)))
+        enum isAllZeroBits = value is 0;
+    else static if (isStaticArray!(typeof(value)))
+    {
+        static if (value.length == 0)
+            enum isAllZeroBits = true;
+        else
+        {
+            alias E = typeof(value[0]);
+            enum isAllZeroBits = mixin(()
+            {
+                enum N = typeof(value).length;
+                char[] buffer;
+                buffer ~= "isAllZeroBits!(E,value[0])";
+                import std.conv : to;
+                foreach (i; 1 .. N)
```

Comment:
Because it could have been either I avoided it due to [seeing this advice on the forums](http://forum.dlang.org/post/mailman.1994.1515288062.9493.digitalmars-d-learn@puremagic.com):
>When you can use the old style do so. Since it puts less stress on the compiler in the general case.

Reply:
That's bad advice. There isn't any measurable performance penality - I tried a large-scale test: https://github.com/dlang/phobos/pull/5989